UNREACHABLE executed at lib/SILOptimizer/Mandatory/MoveOnlyWrappedTypeEliminator.cpp:56!

### Description

crashes on main and on 5.10

### Reproduction

```swift
struct ErrorDescription
{
    let caught:ContinuousClock.Instant
    let type:any Error.Type
    let dump:String

    init(caught:ContinuousClock.Instant, type:any Error.Type, dump:String)
    {
        self.caught = caught
        self.type = type
        self.dump = dump
    }
}
extension ErrorDescription
{
    init(error:borrowing any Error)
    {
        self.init(caught: .now,
            type: Swift.type(of: error),
            dump: String.init(reflecting: error))
    }
}

```


### Stack dump

```text
Unhandled SIL Instruction:   %12 = existential_metatype $
@thick any Error.Type, %5 : $any Error // user: %21
error
UNREACHABLE executed at /home/build-user/swift/lib/SILOptimizer/Mandatory/MoveOnlyWrappedTypeEliminator.cpp:56!
Stack dump:
0.      Program arguments: /usr/bin/swift-frontend -frontend -interpret - -disable-objc-interop -color-diagnostics -enable-bare-slash-regex -empty-abi-descriptor -Xcc -working-directory -Xcc /app -resource-dir /usr/lib/swift -module-name main -plugin-path /usr/lib/swift/host/plugins -plugin-path /usr/local/lib/swift/host/plugins
1.      Swift version 6.0-dev (LLVM 57177aa1b91540b, Swift 8be62863326595c)
2.      Compiling with effective version 5.10
3.      While evaluating request ExecuteSILPipelineRequest(Run pipelines { Mandatory Diagnostic Passes + Enabling Optimization Passes } on SIL for main)
4.      While running pass #65 SILFunctionTransform "MoveOnlyTypeEliminator" on SILFunction "@$s4main16ErrorDescriptionV5errorACs0B0_ph_tcfC".
 for 'init(error:)' (at <stdin>:16:5)
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend 0x0000602782ea55a7
1  swift-frontend 0x0000602782ea32fe
2  swift-frontend 0x0000602782ea5c1a
3  libc.so.6      0x0000754cd1c92520
4  libc.so.6      0x0000754cd1ce69fc pthread_kill + 300
5  libc.so.6      0x0000754cd1c92476 raise + 22
6  libc.so.6      0x0000754cd1c787f3 abort + 211
7  swift-frontend 0x0000602782dfac6f
8  swift-frontend 0x000060277cf9c7db
9  swift-frontend 0x000060277cf9b1a4
10 swift-frontend 0x000060277cf99d75
11 swift-frontend 0x000060277ca50bb8
12 swift-frontend 0x000060277ca51ea1
13 swift-frontend 0x000060277ca5510c
14 swift-frontend 0x000060277ca4e8b8
15 swift-frontend 0x000060277ca4e86d
16 swift-frontend 0x000060277caa084a
17 swift-frontend 0x000060277ca75d86
18 swift-frontend 0x000060277ca4eaad
19 swift-frontend 0x000060277ca8d961
20 swift-frontend 0x000060277c41b8d7
21 swift-frontend 0x000060277c0ed7e3
22 swift-frontend 0x000060277c0ec9ea
23 swift-frontend 0x000060277c103605
24 swift-frontend 0x000060277c0f0717
25 swift-frontend 0x000060277c0eea9c
26 swift-frontend 0x000060277be9ee6e
27 libc.so.6      0x0000754cd1c79d90
28 libc.so.6      0x0000754cd1c79e40 __libc_start_main + 128
29 swift-frontend 0x000060277be9df15
timeout: the monitored command dumped core
Aborted
```


### Expected behavior

it should not crash

### Environment

Swift version 6.0-dev (LLVM 57177aa1b91540b, Swift 8be62863326595c)
Target: x86_64-unknown-linux-gnu

### Additional information

_No response_

PR to resolve this is here: https://github.com/swiftlang/swift/pull/74706, but do note that instead of crashing, the compiler will now tell you that what you're trying to do is an error:
```swift
/Users/alejandro/4.swift:16:10: error: 'error' is borrowed and cannot be consumed
14 | extension ErrorDescription
15 | {
16 |     init(error:borrowing any Error)
   |          `- error: 'error' is borrowed and cannot be consumed
17 |     {
18 |         self.init(caught: .now,
19 |             type: Swift.type(of: error),
20 |             dump: String.init(reflecting: error))
   |                                           `- note: consumed here
21 |     }
22 | }
```

--- a/lib/SILOptimizer/Mandatory/MoveOnlyWrappedTypeEliminator.cpp
+++ b/lib/SILOptimizer/Mandatory/MoveOnlyWrappedTypeEliminator.cpp
@@ -198,6 +198,7 @@ struct SILMoveOnlyWrappedTypeEliminatorVisitor
   NO_UPDATE_NEEDED(ClassMethod)
   NO_UPDATE_NEEDED(FixLifetime)
   NO_UPDATE_NEEDED(AddressToPointer)
+  NO_UPDATE_NEEDED(ExistentialMetatype)
 #undef NO_UPDATE_NEEDED
 
   bool eliminateIdentityCast(SingleValueInstruction *svi) {

--- a/test/SILOptimizer/moveonly_type_eliminator.sil
+++ b/test/SILOptimizer/moveonly_type_eliminator.sil
@@ -36,6 +36,8 @@ struct KlassPair {
 
 sil @use_klass : $@convention(thin) (@guaranteed Klass) -> ()
 
+protocol Error {}
+
 ///////////
 // Tests //
 ///////////
@@ -598,4 +600,16 @@ bb0(%x : @owned $Klass):
   dealloc_box %box : ${ var @moveOnly Klass }
   %retval = tuple ()
   return %retval : $()
+}
+
+// CHECK-LABEL: sil [ossa] @existential_metatype : $@convention(thin) (@guaranteed any Error) -> @thick any Error.Type {
+// CHECK:         [[EM:%.*]] = existential_metatype $@thick any Error.Type, {{%.*}} : $any Error
+// CHECK:         return [[EM]] : $@thick any Error.Type
+// CHECK-LABEL: } // end sil function 'existential_metatype'
+sil [ossa] @existential_metatype : $@convention(thin) (@guaranteed any Error) -> @thick any Error.Type {
+bb0(%x : @guaranteed $any Error):
+  %y = copyable_to_moveonlywrapper [guaranteed] %x : $any Error
+  %em = existential_metatype $@thick any Error.Type, %y : $@moveOnly (any Error)
+  debug_value %em : $@thick any Error.Type, var, name "s", argno 1, expr op_deref
+  return %em : $@thick any Error.Type
 }

